src/libs/DateUtils.ts

```diff
@@ -730,6 +730,15 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Return the date with full format if the created date is the current date.
+ * Otherwise return the created date.
+ */
+function enrichMoneyRequestTimestamp(created: string): string {
+    const currentTime = getDBTime();
+    return formatWithUTCTimeZone(created) === formatWithUTCTimeZone(currentTime) ? currentTime : created;
```

Comment:
Sorry, I won't happily approve more smelly approaches to date operations. Let's stick with the best thing we got, using `date-fns` as much as possible.

You'll have to convince me _why_ that's not the good idea.

Reply:
Also, is this even correct? Aren't we ultimately forced to use the "DB time" format?

Reply:
I think that we should first ask ourselves what timezone is the original date-only `created` string. As I understand it, it's in the user-local timezone! So we should compare it with the user-local "now".

>  I think isToday function is helpful for us in this case.

Also what I started with, but likely it's not a clean solution. It, likely, will have a problem of time passing between two invocations of `now()` (`new Date()`).

Reply:
I sketched the best solution I could think of [here](https://github.com/Expensify/App/pull/34784#discussion_r1463029210). Let me know if it's unclear or if you disagree with the approach.

Reply:
I'm confused, those strings don't have a timezone marker. How does the backend know what is the timezone context of a string like "YYYY-MM-DD HH:MM:SS"?

Reply:
We shouldn't perform date/time algorithms on the string level, unless absolutely necessary. Especially with more code like "let's replace Ts and Zs with `' '` and we'll be fine". Let's not make this code worse than it already is.
 
Please explain to me what's wrong with this approach:

```ts
const now = new Date();
return isSameDate(new Date(created), now) ? getDBTimeFromDate(now) : created;
```

Reply:
Okey, let's try...

```ts
const now = new Date();
const createdDate = parse(created, CONST.DATE.FNS_FORMAT_STRING, now);
return isSameDate(createdDate, now) ? getDBTimeFromDate(now) : created;
```

Reply:
Time zones are the worst. Just to clarify, all datetimes that get saved to our back end should be stored as UTC strings in the `YYYY-MM-DD HH:MM:SS` format. But the kicker is that when displaying them back to the user, we probably need to show it relative to their time zone.

Reply:
Timezones are the worst, true, but we're not helping ourselves by making very questionable decisions like passing dates around as strings, converting `Date`s to string to `Date`s to strings to `Date`..., not using separate types for instants and dates and datetimes, etc., etc.